vignettes/sparsegl.Rmd

```diff
@@ -0,0 +1,332 @@
+predictors of dimension n-obs by p-feature. To initiate y, namely a vector of real-value 
+responses of length n, or a matrix with 1 colnmn with
+$$
+y = X\beta^* + \epsilon,
+$$
+where the coefficient vector $\beta^*$ is specified as below, and the white noise
+$\epsilon$ following standard normal distribution serves as data variation.
+
+```{r}
+library(sparsegl)
+set.seed(1010)
+n <- 100
+p <- 200
+X <- matrix(data = rnorm(n*p, mean = 0, sd = 1), nrow = n, ncol = p)
+eps <- rnorm(n, mean = 0, sd = 1)
+beta_star <- c(rep(5, 5), c(5, -5, 2, 0, 0), rep(-5, 5), c(2, -3, 8, 0, 0), rep(0, (p - 20)))
+y <- X %*% beta_star + eps
+groups <- rep(1:(p / 5), each = 5)
+```
+
+## Function `sparsegl()`
+
+`sparsegl()` is the main function in the `sparsegl` package. Given an input matrix 
+X, and a response vector y (or a matrix with 1 column), a sparse group-lasso 
+regularized linear model is fit for a sequence of penalty parameter values in 
+terms of penalized maximum likelihood. Other arguments of this function can be 
+optionally initialized differently than the default values, and the main arguments are:
+
+* `group`: a vector with consecutive integers of length p indicating the grouping 
+of the features. By default, each group only contains one feature if without initialization.
+
+* `nlambda`: the number of lambdas to use in the regularization path, and default
+is 100.
+
+* `lambda.factor`: the ratio of smallest lambda and the largest lambda in the sequence
+of `lambda`. By default, it is determined by the number of observations and the 
+number of features.
+
+* `lambda`: a user supplied lambda sequence. It is recommended to be not initialized,
+and the program can determine it in terms of `nlambda` and `lambda.factor`.
+
+* `pf`: a penalty vector of length of the total number of groups bn. Each element of 
+this vector is applied to corresponding group size bs and its default value is 
+$\sqrt{\text{bs}}$. 
+
+* `dfmax`: the maximum number of groups in the model. Default is bn + 1.
+
+* `pmax`: the maximum number of groups to be zero. Default is 
+$\min($`dfmax` * 1.2, bn).
+
+* `asparse`: the weight of lasso penalty and default is 0.05. `asparse` = 1 is
+the lasso penalty. `asparse` = 0 is the group lasso penalty.
+
+* `intercept`: indicate if the intercept is included and default is TRUE.
+
+* `standardize`: indicate if the features are standardized ahead and default is TRUE.
+
+The useful information that this function returns is:
+
+* `group`: a vector with consecutive integers of length p indicating the grouping 
+of the features.
+
+* `b0`: the intercept sequence of length `nlambda` if `intercept` = TRUE.
+
+* `beta`: a p-by-`nlambda` matrix of coefficients.
+
+* `df`: the number of nonzero groups for each lambda value.
+
+* `lambda`: the actual lambda sequence of length `nlambda` used in the program.
+
+
+```{r}
+fit1 <- sparsegl(X, y, group = groups)
+summary(fit1)
+```
+
+## Function `plot.sparsegl()`
```

Comment:
`plot`, `coef`, `predict` and `print` are the methods. I would describe them together. "The package provides methods to ..." Then describe anything that's different about these relative to expectation. What does plot do? It's the path but the x-axis can be two different things, and the y axis is either the group norm or the coefficient. `coef` returns a matrix unless you pick a `lambda`. Same with `predict`.